lib/beautiful/init.lua

```diff
@@ -113,14 +115,15 @@ local function load_font(name)
             name = name:to_string()
         end
     end
-    if fonts[name] then
-        return fonts[name]
+    local fonts_key = {name, screen}
+    if fonts[fonts_key] then
+        return fonts[fonts_key]
```

Comment:
Whoops. I totally forgot about this function ignoring DPI.

Anyway, the above cache check does not work. Lua tables are only equal to themselves, i.e. the index operator does not do a deep comparison:
```
$ lua -e 't={} t[{}] = 42 print(t[{}])'
nil
```
Thus, the above doesn't cache check does not work.

Also, I wonder if it makes sense to have a version of this function that gets DPI as argument and another that gets a screen object, gets the screen's DPI and then just calls the other function (same for `get_font_height`). Thoughts?

(At least the cache only needs the DPI, not the screen itself)